### ISSUE TITLE
Make ActionResult handling behind `EnableActionResultHandling` the default in 3.0

### ISSUE THREAD

**Issue by fabiocav** [Azure/azure-functions-host#4907]
For backwards compatibility reasons, some `ActionResult` handling enhancements added to 2.0 had to be placed behind a feature flag (`EnableActionResultHandling`). 

We may want to consider a flag to maintain the 2.0 behavior if we believe that would ease migration, but the behavior behind the flag should be the default in 3.0.

**Comment by brettsam** [comment]
@fabiocav -- what's the deal with this proposal? Is this something we could/should do in v4?

**Comment by ericsampson** [comment]
> 
> 
> @fabiocav -- what's the deal with this proposal? Is this something we could/should do in v4?

yes please. Along with making `DisableDevModeInDebug` the default behavior as well.

**Comment by fabiocav** [comment]
Will create breaking change proposals for both items.

**Comment by ericsampson** [comment]
Is `EnableEnhancedScopes` getting rolled out by default soon? That one seems more of a bug fix than change, I can't see people wanting the original behavior.
